### PR TITLE
PUB-2588 Add List Type For Court Subscription 

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
### JIRA link

[https://tools.hmcts.net/jira/browse/<PR-name-here>](https://tools.hmcts.net/jira/browse/PUB-2588)

### Change description

Add List Type For Court Subscription 

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

**Review by NatashaAlker — APPROVED** [review]

**Review by NatashaAlker — DISMISSED** [review]

File: src/main/views/subscription-add-list.njk [hunk 1]
```diff
@@ -0,0 +1,102 @@
+{% extends "template.njk" %}
+{% from "./macros/common-components.njk" import goBack, submitButton %}
+{%- from "govuk/components/checkboxes/macro.njk" import govukCheckboxes -%}
+{%- from "govuk/components/input/macro.njk" import govukInput -%}
```
Comment (KianKwa) [inline src/main/views/subscription-add-list.njk:4]: if govukCheckboxes and govukInput nunjucks macro are not used we can remove them here

File: src/main/views/subscription-add-list-language.njk [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+{% from "govuk/components/back-link/macro.njk" import govukBackLink %}
+{% from "govuk/components/error-summary/macro.njk" import govukErrorSummary %}
+{% from "govuk/components/radios/macro.njk" import govukRadios %}
+
+{% block pageTitle %}
+    {{ title }}
+{% endblock %}
+
+{% block beforeContent %}
+    {{ govukBackLink({
```
Comment (KianKwa) [inline src/main/views/subscription-add-list-language.njk:12]: Can we not use the goBack() macro defined in common-components here?

File: src/main/resources/locales/en/pending-subscriptions.json [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
     "title": "Confirm your email subscriptions",
+    "title_court_subscription": "Your email subscriptions",
     "removeText": "Remove",
     "table": {
         "header1": "Case name",
         "header2": "Party name(s)",
         "header3": "Reference number",
         "header4": "Actions"
     },
+    "buttonText_court_subscription": "Continue",
```
Comment (KianKwa) [inline src/main/resources/locales/en/pending-subscriptions.json:11]: title_court_subscription and buttonText_court_subscription need to be added to the welsh language resource too

File: src/main/service/LocationService.ts [hunk 4]
```diff
@@ -99,4 +99,14 @@ export class LocationService {
     public async deleteLocationById(locationId: number, requester: string): Promise<object> {
         return await locationRequest.deleteCourt(locationId, requester);
     }
+
+    public async findCourtsJurisdiction(locations): Promise<object> {
```
Comment (KianKwa) [inline src/main/service/LocationService.ts:103]: Can we change the returned type to Promise<string[]> and not object?

File: src/main/routes/routes.ts [hunk 5]
```diff
@@ -282,6 +282,17 @@ export default function (app: Application): void {
         isPermittedMedia,
         app.locals.container.cradle.subscriptionManagementController.get
     );
+    app.get('/subscription-add-list', isPermittedMedia, app.locals.container.cradle.subscriptionAddListController.get);
```
Comment (KianKwa) [inline src/main/routes/routes.ts:285]: We need to add new a11y tests for the new routes too

File: src/main/views/subscription-add-list.njk (hunk 1 above)
Comment (junaidiqbalmoj) [reply src/main/views/subscription-add-list.njk]: removed.

File: src/main/views/subscription-add-list-language.njk (hunk 2 above)
Comment (junaidiqbalmoj) [reply src/main/views/subscription-add-list-language.njk]: as discussed, updated.

File: src/main/resources/locales/en/pending-subscriptions.json (hunk 3 above)
Comment (junaidiqbalmoj) [reply src/main/resources/locales/en/pending-subscriptions.json:11]: added

File: src/main/service/LocationService.ts (hunk 4 above)
Comment (junaidiqbalmoj) [reply src/main/service/LocationService.ts]: done, thanks, good catch :)

File: src/main/routes/routes.ts (hunk 5 above)
Comment (junaidiqbalmoj) [reply src/main/routes/routes.ts:285]: added.

**Review by KianKwa — DISMISSED** [review]

**Pull request closed without merging**